src/common/com/intellij/plugins/haxe/ide/annotator/HaxeAnnotationHolder.java

```diff
@@ -49,7 +55,22 @@ public HaxeAnnotationHolder(AnnotationHolder underlyingHolder) {
 
   public Annotation addAnnotation(HaxeAnnotation haxeAnnotation) {
     if (null == haxeAnnotation) return null;
+    Annotation preexisting = findAnnotation(haxeAnnotation.getSeverity(),
+                                            haxeAnnotation.getRange(),
+                                            haxeAnnotation.getMessage());
+    if (null != preexisting) {
```

Comment:
In this file, you will see us stashing the annotations that we create into the annotation session and not creating duplicates.  This is annoying because IDEA used to remove duplicates themselves.  Now they do not, but they do not allow duplicate error markings in the test input.  So, the answer was to not produce duplicates in the first place.

Unfortunately, the way the annotation system is designed, all of the annotations that we place in an AnnotationHolder are copied out to another data structure and then the holder cleared between PsiElements.  That means that we can't use the existing data structures to detect what we've already annotated.  Instead, we have to do this job ourselves.

I could have taken the direction of creating Haxe-specific versions of the highlighting algorithms (GeneralHighlightingPass -- which disappears in 2019.2 -- and DefaultHighlightVisitor), but they are fairly complex and that would have taken quite a bit of code, and some trickery to pull off.  Just stashing the annotations was far more straightforward.